akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ShardingDirectReplication.scala

```diff
@@ -62,17 +66,25 @@ private[akka] object ShardingDirectReplication {
 
       Behaviors.receiveMessage {
         case WrappedPublishedEvent(event) =>
-          context.log.trace(
-            "Forwarding event for persistence id [{}] sequence nr [{}] to replicas",
-            event.persistenceId,
-            event.sequenceNumber)
-          replicaShardingProxies.foreach {
-            case (replica, proxy) =>
-              val newId = ReplicationId.fromString(event.persistenceId.id).withReplica(replica)
-              val envelopedEvent = ShardingEnvelope(newId.persistenceId.id, event)
-              if (!selfReplica.contains(replica)) {
-                proxy.asInstanceOf[ActorRef[ShardingEnvelope[PublishedEvent]]] ! envelopedEvent
+          try {
+            val replicationId = ReplicationId.fromString(event.persistenceId.id)
+            if (replicationId.typeName == typeName) {
+              context.log.traceN(
+                "Forwarding event for persistence id [{}] sequence nr [{}] to replicas.",
+                event.persistenceId,
+                event.sequenceNumber)
+              replicaShardingProxies.foreach {
+                case (replica, proxy) =>
+                  val newId = ReplicationId.fromString(event.persistenceId.id).withReplica(replica)
+                  val envelopedEvent = ShardingEnvelope(newId.persistenceId.id, event)
+                  if (!selfReplica.contains(replica)) {
+                    proxy.asInstanceOf[ActorRef[ShardingEnvelope[PublishedEvent]]] ! envelopedEvent
+                  }
               }
+            }
+          } catch {
+            case _: IllegalArgumentException =>
+            // ignore event as persistence id not from replicated event sourcing
```

Comment:
Since constructing one exception with stack trace for each event is expensive perhaps it should, log a warning at least once, add something to a cheaper blacklist, or have a `ReplicationId.isReplicationId(string)`?

Reply:
👍